dynamiqs/integrators/core/montecarlo_integrator.py

```diff
@@ -0,0 +1,44 @@
+import equinox as eqx
+from jaxtyping import Array
+
+from ...result import Result, SolveSaved
+from ..apis.jssesolve import _vectorized_jssesolve
+from .abstract_integrator import BaseIntegrator
+from .interfaces import MEInterface, SolveInterface
+
+
+class JumpMonteCarloExtra(eqx.Module):
+    clicktimes: Array
+    final_state_norm: Array
+
+
+class MESolveJumpMonteCarloIntegrator(BaseIntegrator, MEInterface, SolveInterface):
+    def run(self) -> Result:
+        # call _vectorized_jssesolve to compute the jump SSE results
+        jsse_result = _vectorized_jssesolve(
+            self.H,
+            self.Ls,
+            self.y0,
+            self.ts,
+            self.method.keys,
+            self.Es,
+            self.method.jsse_method,
+            self.gradient,
+            self.method.jsse_options,
+        )
+
+        # compute saved with the mean results
+        saved = SolveSaved(
+            ysave=jsse_result.mean_states,
+            Esave=jsse_result.mean_expects,
+            extra=JumpMonteCarloExtra(
+                clicktimes=jsse_result.clicktimes,
+                final_state_norm=jsse_result.final_state_norm,
+            ),
```

Comment:
If we want to save this it should be in infos, not extra. `result.extra` is for
> Extra data saved with `save_extra()` if specified in `options`.

Now:
- I don't think we want to add these two bits to the infos, then the user should _really_ use `jssesolve`.
- We should add back the `save_extra` mechanism, or document that it doesn't work.